main/zookeeper_watcher.go

```diff
@@ -0,0 +1,207 @@
+	Timeout            int64  `env:"WATCHER_TIMEOUT" envDefault:"60"`
+}
+
+type ZookeeperBroker struct {
+	Path       string
+	Connection *zk.Conn
+	PathExists bool
+}
+
+func (broker *ZookeeperBroker) WriteMessage(message []byte) error {
+	timestamp := time.Now().UTC().Format(time.RFC3339Nano)
+	flags := int32(0)
+	acl := zk.WorldACL(zk.PermAll)
+	if !broker.PathExists {
+		data, stat, err := broker.Connection.Get(broker.Path)
+		log.Infof("Read %s %s %s %s", broker.Path, data, stat, err)
+		if err != nil && err == zk.ErrNoNode {
+			_, err := broker.Connection.Create(broker.Path, nil, flags, acl)
+			if err != nil {
+				log.Errorf("Creating %s failed: %s", broker.Path, err)
+				return err
+			}
+		}
+		broker.PathExists = true
+	}
+	messagePath := fmt.Sprintf("%s/%s", broker.Path, timestamp)
+
+	log.Infof("Trying to create %s", messagePath)
+	path, err := broker.Connection.Create(messagePath, message, flags, acl)
+	log.Infof("Create returned %s %s", path, err)
+
+	if err != nil {
+		return err
+	}
+	if path != messagePath {
+		// Try to delete it since it did not go where we expected
+		log.Errorf("Tried to create node at %s but was created at %s", messagePath, path)
+		broker.Connection.Delete(path, -1)
+		return fmt.Errorf("Zookeper Node Creation Failure")
+	}
+	log.Infof("Successfully created %s", messagePath)
+	return nil
+}
+
+func (broker *ZookeeperBroker) ReadMessage(timeout time.Duration) (*[]byte, error) {
+	start := time.Now()
+	for time.Since(start) < timeout {
+		children, _, err := broker.Connection.Children(broker.Path)
+		if len(children) == 0 {
+			log.Infof("No children in %s", broker.Path)
+			delay := timeout / 10
+			if time.Since(start) > (timeout + delay) {
+				break
+			}
+			time.Sleep(delay)
+			continue
+		}
+		log.Infof("Found %s in %s", children, broker.Path)
+		sort.Strings(children)
+		first := children[0]
+		messagePath := fmt.Sprintf("%s/%s", broker.Path, first)
+		data, stat, err := broker.Connection.Get(messagePath)
+		if err != nil {
+			return nil, err
+		}
+		err = broker.Connection.Delete(messagePath, stat.Version)
+		if err != nil {
+			log.Errorf("Failed to delete old message at %s", messagePath)
+			// Nothing really to do so go on
+		}
+		return &data, nil
+	}
+
+	// Timeout
+	return nil, fmt.Errorf("No data after %s", time.Since(start))
+}
+
+func main() {
+	configuration := watcherConfiguration{}
+	err := configEnv.Parse(&configuration)
+	if err != nil {
+		log.Fatalf("Failed to parse envrionment variables. %s", err)
+	}
+
+	healthCheckPath := configuration.HealthCheckPath
+	if len(healthCheckPath) < 0 {
+		log.Fatalf("Invalid HEALTH_CHECK_PATH, must not be empty")
+	}
+	if !strings.HasPrefix(healthCheckPath, "/") {
+		healthCheckPath = "/" + healthCheckPath
+	}
+	zookeeperServers := configuration.ZookeeperServers
+	if len(zookeeperServers) < 0 {
+		log.Fatalf("Invalid ZOOKEEPER_SERVERS, must not be empty")
+	}
+	prometheusEndpoint := configuration.PrometheusEndpoint
```

Comment:
why do you check all other variables are set correctly besides this one?

Reply:
Done